Controller/ThemeController.php

```diff
@@ -0,0 +1,219 @@
+<?php
+
+/*
+ * This file is part of the Integrated package.
+ *
+ * (c) e-Active B.V. <integrated@e-active.nl>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Integrated\Bundle\ThemeBundle\Controller;
+
+use Integrated\Bundle\ThemeBundle\Form\Type\ThemeEditorType;
+use Integrated\Common\Channel\Connector\Adapter\RegistryInterface;
+use Integrated\Common\Channel\Connector\Config\ConfigManagerInterface;
+
+use Symfony\Bundle\FrameworkBundle\Controller\Controller;
+
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\HttpFoundation\Request;
+
+use Symfony\Component\Finder\Finder;
+
+use Integrated\Bundle\ThemeBundle\Entity\ThemePath;
+
+/**
+ * @author Jan Sanne Mulder <jansanne@e-active.nl>
+ */
+class ThemeController extends Controller
+{
+    /**
+     * @var ConfigManagerInterface
+     */
+    protected $manager;
+
+    /**
+     * @var RegistryInterface
+     */
+    protected $registry;
+
+    /**
+     * Constructor/
+     *
+     * @param ConfigManagerInterface $manager
+     * @param RegistryInterface      $registry
+     * @param ContainerInterface     $container
+     */
+    public function __construct(
+        ConfigManagerInterface $manager,
+        RegistryInterface $registry,
+        ContainerInterface $container
+    ) {
+        $this->manager = $manager;
+        $this->registry = $registry;
+
+        $this->container = $container;
+    }
+
+    /**
+     * @return \Symfony\Component\HttpFoundation\Response
+     */
+    public function indexAction()
+    {
+        $themeManager = $this->get('integrated_theme.templating.theme_manager');
+
+        $routeId = [];
```

Comment:
Logic 67 til 72 can be put in a class (constructor takes $themeManager and has one public get method